WalletWasabi/Services/WalletService.cs

```diff
@@ -719,6 +720,45 @@ public void RenameLabel(SmartCoin coin, SmartLabel newLabel)
 			}
 		}
 
+		internal static LockTime InternalSelectLockTimeForTransaction(uint tipHeight)
+		{
+			try
+			{
+				// We use the timelock distribution observed in the bitcoin network
+				// in order to reduce the wasabi wallet transactions fingerprinting
+				// chances.
+				// 
+				// Network observations:
+				// 90.0% uses locktime = 0
+				//  7.5% uses locktime = current tip
+				//  0.65% uses locktime = next tip (current tip + 1)
+				//  1.85% uses up to 5 blocks in the future (we don't do this)
+				//  0.65% uses an uniform random from -1 to -99
+
+				var randomGenerator = new Random();
```

Comment:
Use one (static?) random instance.

Reply:
Why? what's the point?

Reply:
https://stackoverflow.com/a/2643442/2061103

Reply:
> It is best practice to create a single instance of Random and use it throughout your program - otherwise the results may not be as random.